lib/util.js

```diff
@@ -282,612 +365,515 @@ inspect.styles = Object.assign(Object.create(null), {
+    const extra = [];
+    // Only handle special keys
+    var key;
+    for (i = keys.length - 1; i >= 0; i--) {
+      key = keys[i];
+      if (numberRegExp.test(key) && +key < 2 ** 32 - 1)
+        break;
+      extra.push(formatProperty(ctx, value, recurseTimes, key, 2));
     }
+    for (i = extra.length - 1; i >= 0; i--)
+      output.push(extra[i]);
   }
   return output;
 }
 
+function formatArray(ctx, value, recurseTimes, keys) {
+  const len = Math.min(Math.max(0, ctx.maxArrayLength), value.length);
+  const valLen = value.length;
+  const keyLen = keys.length - (ctx.showHidden ? 1 : 0);
+  if (keyLen !== valLen || keys[keyLen - 1] !== `${valLen - 1}`)
+    return formatSpecialArray(ctx, value, recurseTimes, keys, len, valLen);
 
-function formatTypedArray(ctx, value, recurseTimes, visibleKeys, keys) {
+  const output = [];
+  for (var i = 0; i < len; i++)
+    output.push(formatProperty(ctx, value, recurseTimes, keys[i], 1));
+  const remaining = valLen - i;
+  if (remaining > 0)
+    output.push(`... ${remaining} more item${remaining > 1 ? 's' : ''}`);
+  if (ctx.showHidden === true)
+    output.push(formatProperty(ctx, value, recurseTimes, 'length', 2));
+  return output;
+}
+
+function formatTypedArray(ctx, value, recurseTimes, keys) {
   const maxLength = Math.min(Math.max(0, ctx.maxArrayLength), value.length);
   const remaining = value.length - maxLength;
-  var output = new Array(maxLength);
-  for (var i = 0; i < maxLength; ++i)
-    output[i] = formatNumber(ctx, value[i]);
+  const output = new Array(maxLength);
+  var i;
+  for (i = 0; i < maxLength; ++i)
+    output[i] = formatNumber(ctx.stylize, value[i]);
   if (remaining > 0) {
     output.push(`... ${remaining} more item${remaining > 1 ? 's' : ''}`);
   }
-  for (const key of keys) {
-    if (typeof key === 'symbol' || !numbersOnlyRE.test(key)) {
-      output.push(
-        formatProperty(ctx, value, recurseTimes, visibleKeys, key, true));
+  if (ctx.showHidden) {
+    // .buffer goes last, it's not a primitive like the others.
+    const extraKeys = [
+      'BYTES_PER_ELEMENT',
+      'length',
+      'byteLength',
+      'byteOffset',
+      'buffer'
+    ];
+    for (i = 0; i < extraKeys.length; i++) {
+      const str = formatValue(ctx, value[extraKeys[i]], recurseTimes);
+      output.push(`[${extraKeys[i]}]: ${str}`);
     }
   }
+  // TypedArrays cannot have holes. Therefore it is safe to assume that all
+  // extra keys are indexed after value.length.
+  for (i = value.length; i < keys.length; i++) {
+    output.push(formatProperty(ctx, value, recurseTimes, keys[i], 2));
+  }
   return output;
 }
 
-
-function formatSet(ctx, value, recurseTimes, visibleKeys, keys) {
-  var output = [];
-  value.forEach(function(v) {
-    var nextRecurseTimes = recurseTimes === null ? null : recurseTimes - 1;
-    var str = formatValue(ctx, v, nextRecurseTimes);
-    output.push(str);
-  });
+function formatSet(ctx, value, recurseTimes, keys) {
+  const output = [];
```

Comment:
Same here, this could be made a "fixed array."

Reply:
See above.